core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -245,6 +244,14 @@ protected void doStop() throws Exception {
       context.setBaseResource(staticResources);
     }
 
+    if(config.getBoolean(RestConfig.ENABLE_GZIP_COMPRESSION_CONFIG)) {
+      FilterHolder gzipFilter = new FilterHolder(AsyncGzipFilter.class);
+      // do not check if .gz file already exists for the requested resource
+      gzipFilter.setInitParameter("checkGzExists", "false");
+      gzipFilter.setInitParameter("methods", "GET,POST");
+      context.addFilter(gzipFilter, "/*", null);
```

Comment:
Nothing specific, but since this covers everything it seems like it'd be potentially bad. For example, if you're serving static assets, gzipping a png is just going to be a waste of time. I had also been thinking of other already compressed data, e.g. in the REST proxy, but today the compressed form isn't exposed anyway, so maybe not worth worrying about that case.

Fine to leave it as is, was just calling out this as a potential limitation since it might change how we'd want to represent the config options.